client/my-sites/plans/navigation.jsx

```diff
@@ -93,13 +95,18 @@ class PlansNavigation extends React.Component {
 						{ canManageDomain && (
 							<NavItem
 								path={ `/domains/manage/${ site.slug }` }
+								onClick={ () => this.props.dispatch( expandSection( SIDEBAR_SECTION_MANAGE ) ) }
```

Comment:
Instead of calling dispatch directly, we can map this in mapDispatchToProps:

https://react-redux.js.org/using-react-redux/connect-mapdispatch

Usually looks a bit like::

Using mapDispatchToProps object shorthand:
```jsx
// in class
expandManageSection  = () => this.props.expandSection( SIDEBAR_SECTION_MANAGE );

// render
onClick={ this.expandManageSection }

connect( 
    mapStateToProps, 
    { expandSection }
)( localize( PlansNavigation ) );
```

Or using the function signature of mapDispatchToProps:
```jsx
//... in class render:
onClick={ this.props.expandManageSection }

//...
const mapDispatchToProps = ( dispatch, ownProps ) => {
  return {
    expandManageSection: () => dispatch( expandSection( SIDEBAR_SECTION_MANAGE ) );
  }
}

connect( 
    mapStateToProps, 
    mapDispatchToProps,
)( localize( PlansNavigation ) );
```

Reply:
Thanks for taking the time to write document this.

So the router is handling the expansion of the sections on hard refresh, but not on page change  🤔 Maybe this is something we should look into? As it would prevent dispatching the `expandSection` altogether... 

Yes, encapsulating it makes it more declarative. As I wasn't sure it was the best approach, I didn't really take the time to do it properly. I get on it now 👍

Reply:
> Maybe this is something we should look into?

Fine to investigate on the page.js routing layer too! Let us know if you need help

Reply:
Depending on how the subsection header is built, we might be able to take advantage of navigation middleware to perform this as a side-effect as well, if it dispatches the NAVIGATION redux event instead of calling page directly.

https://github.com/Automattic/wp-calypso/blob/89f92a4c795971a866a3aaac4dd88cf9ee75e97e/client/state/navigation/middleware.js#L41

So maybe:

```jsx
case NAVIGATE:
    if ( pathMatchesSidebarSection ) {
        dispatch( expandMySitesSidebarSection( myExampleSection ) );
    }
    if ( action.path ) {
       //page.js navigation side-effect
       page( action.path );
    }
```

Reply:
I pushed a commit with the wrapper for `mapDispatchToProps` for now.

I briefly look into it and couldn't find the logic that sets the menu section as `expanded` anywhere. I am also not very familiar with `pages.js`... So any guidance would help 😄

Reply:
Before messing with the router, I wanted to find out where is the logic to expand the menu when we do a hard refresh. I cannot seem to find it...

Reply:
[Found it ](https://github.com/Automattic/wp-calypso/blob/89f92a4c795971a866a3aaac4dd88cf9ee75e97e/client/layout/sidebar/item.jsx#L50). Basically, `SidebarItem` will trigger expandSection after mounting if selected.

Reply:
So I found a nice workaround, much more appropriate than rigging an expandStecion on the NavigateItem...

Adding this
```
	componentDidUpdate(prevProps) {
		if (prevProps.selected !== this.props.selected && this.props.selected) {
			const { expandSection, selected } = this.props;

			if ( isFunction( expandSection ) && selected ) {
				expandSection();
			}
		}
	  }
```

To the [Sidebar item](https://github.com/Automattic/wp-calypso/blob/89f92a4c795971a866a3aaac4dd88cf9ee75e97e/client/layout/sidebar/item.jsx#L46) will do the trick. Basically making him the owner of managing the expansion of its section depending on its state...